test/unit/specs/core/executeCommandFactory.spec.js

```diff
@@ -101,6 +101,30 @@ describe("executeCommandFactory", () => {
     });
   });
 
+  it("executes component commands", () => {
+    const runCommandSpy = jasmine.createSpy();
+    const testCommand = {
+      run: runCommandSpy,
+      optionsValidator: () => "options"
+    };
+    const componentRegistry = {
+      getCommand: () => testCommand,
+      getCommandNames() {
+        return ["test"];
+      }
+    };
+    const configureCommand = () => Promise.resolve(componentRegistry);
+    const executeCommand = executeCommandFactory({
+      logger,
+      configureCommand,
+      handleError
+    });
+    executeCommand("configure");
+    executeCommand("test").then(() => {
+      expect(runCommandSpy.toHaveBeenCalledWith("option"));
```

Comment:
I believe this should be failing because `option` !== `options`. I think the test isn't waiting for your promise because you're not returning it. Hot tip: always check that your test can fail before making it pass.

Reply:
Hah, I actually changed that to force fail it to verify that it worked